salt/states/boto_ec2.py

```diff
@@ -1752,5 +1749,169 @@ def private_ips_absent(name, network_interface_name=None, network_interface_id=N
 
         # there were no changes since we did not attempt to remove ips
         ret['changes'] = {}
+    return ret
+
+
+def secgroup_present(
```

Comment:
It's inconsistent with the rest of the modules. The way the modules are meant to be designed is that everything related to a single resource is managed through a single state, so for instance, this is bad:

```
Ensure ec2 instance is created:
  boto_ec2.instance_present:
    - name: myec2instance

Ensure myec2instance has security groups:
  boto_ec2.secgroup_present:
    - name: myec2instance
    - security_groups:
       - secgroup1
       - secgroup2
```

This is good:

```
Ensure ec2 instance is created:
  boto_ec2.instance_present:
    - name: myec2instance
    - security_groups:
       - secgroup1
       - secgroup2
```

Folks have added extra functions in here to manage resources that aren't fully salt managed, which isn't ideal, but we accept them because sometimes folks need these as part of a transition.

It's fine to have these functions, as long as they aren't the primary way of maintaining parts of an ec2 instance.